### PR TITLE
SCB-333 Update to support the date time with JSR-310

### PR DESCRIPTION
**Pull request by zhfeng**
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/SCB) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[SCB-XXX] Fixes bug in ApproximateQuantiles`, where you replace `SCB-XXX` with the appropriate JIRA issue.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean install` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

File: foundations/foundation-common/src/test/java/org/apache/servicecomb/foundation/common/utils/TestJsonUtils.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+public class TestJsonUtils {
+    @Test
+    public void testJSR310() throws Exception {
+        String period = "\"P1Y1M2D\"";
+        String duration = "\"P1DT1H2M3S\"";
+        String offsetDateTime = "\"2018-05-28T11:00:00+08:00\"";
+        String localDate = "\"2018-05-28\"";
+        String localDateTime = "\"2018-05-28T11:00:00\"";
+
+        JsonUtils.OBJ_MAPPER.readValue(period, Period.class);
```
Comment (WillemJiang) [inline foundations/foundation-common/src/test/java/org/apache/servicecomb/foundation/common/utils/TestJsonUtils.java]: I think we need to verify the value of object here.

Comment (zhfeng) [reply foundations/foundation-common/src/test/java/org/apache/servicecomb/foundation/common/utils/TestJsonUtils.java]: Ok, it makes sense.

**Comment by wujimin** [comment]
1.RestObjectMapper is for RESTful codec, it's better to add UT for it.
2.this is just for codec, i'm not sure if we can generate swagger from JSR-310 dataTypes
  so it's better to add UT and IT for new dataTypes.

**Comment by zhfeng** [comment]
@wujimin I'm not very sure what the UT and the IT means ?

**Comment by wujimin** [comment]
Unit Test
Integration Test
:)

**Comment by liubao68** [comment]
Integrations tests mean we need to add an end-to-end use case in a RPC consumer & provider. You can add examples in demos/SpringMVCclient and SpringMVCserver. 
Because we will use service definition to generate swagger, and you only changed Json mapper, this PR may not work as we expected.

**Comment by zhfeng** [comment]
So I also need to take a look at the swagger generator codes to make sure it can handle the localDate with the JSR310 ?

**Comment by liubao68** [comment]
Yes, I think so. Or users how to use this feature you provided? I can't figure out what is changed after this PR to developers using java-chassis.

**Comment by imlidian** [comment]
I think I have some pojo codes  for this localDate Type testing. Now I give it a try.
https://github.com/microDevOps/microservice-test

**Comment by WillemJiang** [comment]
@imlidian  Please updated the verification that you did.

**Comment by imlidian** [comment]
The Error and my test case implement way as follows. Did I miss something about adding one test case?

Error:
2018-06-05 10:24:19,605 [WARN] consumer method org.apache.servicecomb.demo.springmvc.client.CodeFirstSpringmvcIntf:testObject not exist in contract. org.apache.servicecomb.swagger.engine.SwaggerEnvironment.createConsumer(SwaggerEnvironment.java:98)

Exception in thread "main" java.lang.IllegalStateException: Consumer method org.apache.servicecomb.demo.springmvc.client.CodeFirstSpringmvcIntf:testObject not exist in contract, microserviceName=springmvc, schemaId=codeFirst; new producer not running or not deployed.

Interface Side:
public interface CodeFirstSpringmvcIntf {
 
   Object testObject(Object input);
 
  LocalDate testLocalDate(LocalDate input);

  //LocalDateTime testLocalDateTime(LocalDateTime input);


}
Server Side:
@RestSchema(schemaId = "codeFirst")
 @RequestMapping(path = "/codeFirstSpringmvc", produces = MediaType.APPLICATION_JSON_VALUE)
 public class CodeFirstSpringmvc {
     return input;
   }
  @PostMapping(path = "/testLocalDate")
  public LocalDate testLocalDate(@RequestBody LocalDate input) {
     return input;
   }

Client Side:
b/demo/demo-springmvc/springmvc-client/src/main/java/org/apache/servicecomb/demo/springmvc/client/TestObject.java
 public class TestObject {
     TestMgr.check("str", result);
     TestMgr.check(String.class, result.getClass());
 
    // LocalDate
    result = intf.testLocalDate(LocalDate.of(1968, 12, 8));
    TestMgr.check(LocalDate.of(1968, 12, 8), result);
    TestMgr.check(LocalDate.class, result.getClass());

    result = restTemplate.postForObject(prefix  "/object", "str", String.class);
    TestMgr.check("str", result);
    TestMgr.check(String.class, result.getClass());
}

**Comment by liubao68** [comment]
You can check your provider logs for detail messages. It usually maybe you add some new methods and microservice version not changed and the consumer cannot see the newly added method from service center.  See here for more details: https://huaweicse.github.io/servicecomb-java-chassis-doc/zh_CN/question-and-answer/interface-compatibility.html

**Comment by liubao68** [comment]
We have opened an API to let users to customize RestObjectMapper https://github.com/apache/incubator-servicecomb-java-chassis/pull/874. If users want to use the feature provided by this PR, they can write their own RestObjectMapper. 
I am going to close this PR since it's not a complete feature. Please feel free to add more comments when you have another solutions.

**Pull request closed without merging**